File: apps/admin_backend/src/modules/sales/entities/cash-movement.entity.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
  Index,
} from 'typeorm';

export enum CashMovementType {
  CASH_IN = 'CASH_IN',
  CASH_OUT = 'CASH_OUT',
  PETTY_CASH = 'PETTY_CASH',
  SAFE_DROP = 'SAFE_DROP',
}

@Entity('cash_movements')
@Index(['tenant_id', 'shift_id'])
@Index(['tenant_id', 'terminal_id'])
export class CashMovement {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column({ type: 'varchar', length: 100 })
  tenant_id: string;

  @Column({ type: 'varchar', length: 100 })
  shift_id: string;

  @Column({ type: 'varchar', length: 100 })
  terminal_id: string;

  @Column({
    type: 'enum',
    enum: CashMovementType,
  })
  type: CashMovementType;

  @Column('decimal', { precision: 12, scale: 4, default: 0 })
  amount_nio: number;

  @Column('decimal', { precision: 12, scale: 4, default: 0 })
  amount_usd: number;

  @Column({ type: 'varchar', length: 255 })
  reason: string;

  @Column({ type: 'varchar', length: 100, nullable: true })
  authorized_by_user_id: string | null;

  @CreateDateColumn({ type: 'timestamptz' })
  timestamp: Date;
}

File: apps/pos_app/lib/data/models/sales/cash_movement_entity.dart
import 'package:floor/floor.dart';

@Entity(tableName: 'cash_movements')
class CashMovementEntity {
  @primaryKey
  final String id;
  @ColumnInfo(name: 'shift_id')
  final String shiftId;
  @ColumnInfo(name: 'terminal_id')
  final String terminalId;
  @ColumnInfo(name: 'type')
  final String type; // 'CASH_IN', 'CASH_OUT', 'PETTY_CASH', 'SAFE_DROP'
  @ColumnInfo(name: 'amount_nio')
  final double amountNio;
  @ColumnInfo(name: 'amount_usd')
  final double amountUsd;
  @ColumnInfo(name: 'reason')
  final String reason;
  @ColumnInfo(name: 'authorized_by_user_id')
  final String? authorizedByUserId;
  @ColumnInfo(name: 'timestamp')
  final int timestamp;
  @ColumnInfo(name: 'sync_status')
  final String syncStatus;

  CashMovementEntity({
    required this.id,
    required this.shiftId,
    required this.terminalId,
    required this.type,
    required this.amountNio,
    this.amountUsd = 0.0,
    required this.reason,
    this.authorizedByUserId,
    required this.timestamp,
    this.syncStatus = 'pending',
  });
}
